### DISCUSSION TITLE
Add middleware

### DISCUSSION THREAD

**Discussion by Melvynx** [richardsolomou/next-safe-route#2] [Ideas]
The library will be perfect when we add support for middleware. I can try to add it if you agree with the feature.

I propose an API like this:

Have a `use` method to add middleware and save routes with middleware configuration like:

```js
const authRoute = createSafeRouter().use(...)
```

Then, you should be able to import "authRoute" in any route file and access the middleware.

For reference, visit https://next-safe-action.dev/docs/safe-action-client/middleware

**Comment by richardsolomou** [comment]
Released with https://github.com/richardsolomou/next-safe-route/releases/tag/0.0.19 - thanks!
